### PR TITLE
ncm-ccm: Clean up

### PR DESCRIPTION
**Pull request by wdpypere**
- Remove pod file
 - move all content of pod file to schema or pm file
 - removed some variables from the schema which have been removed from CCM (context, preprocessor)

### CODE REVIEW

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 1]
```diff
@@ -54,33 +54,57 @@ type kerberos_principal_string = string with {
...
 type component_ccm = {
     include structure_component
+    @{The location of the configuration file. Normally this should not be changed. Defaults to `/etc/ccm.conf`.}
     'configFile'       : string = '/etc/ccm.conf'
+    @{The URL for the machine's profile. You can use either the http or https protocols (the file protocol is also possible eg. for tests). (see ccm-fetch manpage)}
     'profile'          : type_hostURI
+    @{profile failover URL in case the above is not working. (see ccm-fetch manpage)}
     'profile_failover' ? type_hostURI[]
-    'context'          ? type_hostURI
+    @{Turn on debugging. Defaults to 0.}
```
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Don't re-specify defaults in the comment.

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 2]
```diff
@@ -54,33 +54,57 @@ type kerberos_principal_string = string with {
...
     'debug'            : long(0..1) = 0
+    @{Force fetching of the machine profile. Turning this on ignores the modification times. Defaults to 0. }
     'force'            : long(0..1) = 0
-    'preprocessor'     ? string
+    @{The root directory of the CCM cache.  Defaults to `/var/lib/ccm`.}
```
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Ditto.

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 3]
```diff
@@ -54,33 +54,57 @@ type kerberos_principal_string = string with {
...
+    @{The location of the configuration file. Normally this should not be changed. Defaults to `/etc/ccm.conf`.}
     'configFile'       : string = '/etc/ccm.conf'
+    @{The URL for the machine's profile. You can use either the http or https protocols (the file protocol is also possible eg. for tests). (see ccm-fetch manpage)}
     'profile'          : type_hostURI
+    @{profile failover URL in case the above is not working. (see ccm-fetch manpage)}
     'profile_failover' ? type_hostURI[]
-    'context'          ? type_hostURI
+    @{Turn on debugging. Defaults to 0.}
     'debug'            : long(0..1) = 0
+    @{Force fetching of the machine profile. Turning this on ignores the modification times. Defaults to 0. }
```
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Ditto.

File: ncm-ccm/src/main/pan/components/ccm/schema.pan [hunk 4]
```diff
@@ -54,33 +54,57 @@ type kerberos_principal_string = string with {
 
 type component_ccm = {
     include structure_component
+    @{The location of the configuration file. Normally this should not be changed. Defaults to `/etc/ccm.conf`.}
     'configFile'       : string = '/etc/ccm.conf'
+    @{The URL for the machine's profile. You can use either the http or https protocols (the file protocol is also possible eg. for tests). (see ccm-fetch manpage)}
     'profile'          : type_hostURI
+    @{profile failover URL in case the above is not working. (see ccm-fetch manpage)}
```
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: This is actually a list of URLs, I wonder how that is implemented.

File: ncm-ccm/src/main/pan/components/ccm/schema.pan (hunk 1 above)
Comment (wdpypere) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: unfortunately, panc-annotations does not export the defaults.

Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: 😭

Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Opened an issue for this quattor/pan#132

File: ncm-ccm/src/main/pan/components/ccm/schema.pan (hunk 4 above)
Comment (jrha) [inline ncm-ccm/src/main/pan/components/ccm/schema.pan]: Can you update the annotation to state this can be a list of URLs?

**Pull request merged**